### ISSUE TITLE
Improve inline diff highlighting 

### ISSUE THREAD

**Issue by Otiel** [ForkIssues/TrackerWin#226]
Consider the following diff:

![image](https://user-images.githubusercontent.com/3851902/55783637-2818c780-5aaf-11e9-9265-9151f70a9a78.png)

Only the two occurrences of the string `0.2` to `2.1` should be highlighted. The strings circled in purple on the picture shouldn't be highlighted.

**Comment by DanPristupov** [comment]
I tried to implement inline diff based on the Myers (http://blog.robertelder.org/diff-algorithm/) and LCS algorithms (https://en.wikipedia.org/wiki/Longest_common_subsequence_problem), but it's **too** good.

It turned out that **readable** diff != precise diff.

> Only the two occurrences of the string 0.2 to 2.1 should be highlighted.

It's not that easy :).
Let's consider the beginning as an example.
<img width="294" alt="image" src="https://user-images.githubusercontent.com/618115/55786355-5e0c7a80-5ab4-11e9-9649-22c209550f58.png">
As you can notice `.2` combination exists in both `Fody.4.0.2` and `Fody.4.2.1` expressions. So only `.0` (removed) and `.1` (added) must be highlighted :).

Long story short, the precise highlighting becomes more confusing than the the current one.

I'm planning to investigate this topic deeper and try to tweak the algorithm to provide a worse but a more readable result. In a month of few may be.
